### ISSUE TITLE
[BC Transport][Crash] ClassCastException with buildcraft.api.transport.pipe.IPipe

### ISSUE THREAD

**Issue by momocow** [BuildCraft/BuildCraft#4172]
BuildCraft version: `7.99.15`
Forge version: `1.12.2-14.23.2.2651`
Link to crash report or log: `[Pastebin](https://pastebin.com/2Yv323CU)`
Singleplayer or multiplayer:  `multiplayer server`
Steps to reproduce: `unknown`

> java.lang.ClassCastException: net.minecraftforge.items.wrapper.EmptyHandler cannot be cast to buildcraft.api.transport.pipe.IPipe

MP server crashed with ClassCastException.
Seems that EmptyHandler is an exception that should be handled not casting to IPipe.
Just guessing, though.

**Comment by momocow** [comment]
> Description: Exception in server tick loop

Additional info:
I tried to restart my server but failed. It crashed again when loading the world.

**Comment by AlexIIL** [comment]
I think this is caused when a tile entity returns the wrong object from ICapabilityProvider.getCapability.

Can you post the full crash report? It might have more information (like a full mod list).

I've assigned it or myself as i need to add checks for this sort of stuff internally.

**Comment by AlexIIL** [comment]
(Note that this looks very similar to #4061 - without the full crash report I don't actually know if you're using bonsai trees, but seeing as how you are 2 Buildcraft versions behind it seems possible that you might have an outdated bonsai trees as well)

**Comment by momocow** [comment]
Here is the full report.
https://pastebin.com/uGy65Jqk

> bonsaitrees-1.0.3-b67.jar

I saw it lol.
Should update the bonsai tree mod to `1.0.4-b71` fix this problem?

**Comment by AEnterprise** [comment]
yes, please update buildcraft and bonsaitrees, and maybe other mdos that are outdated as well, probably will fix it
